### ISSUE TITLE
Pytorch training on 2D face alignment always fails to converge

### ISSUE THREAD

**Issue by tyshiwo** [1adrianb/face-alignment#83]
Hi,

Recently, I re-implemented the training code of your heatmap model in pytorch since your released code is for inference only. We used Adam/RMSProp as the optimizer, and 300W as the training set. However, we failed the let the model converge and the loss cannot decrease as long as it reaches 0.006. 

We checked the input image and the ground truth coordinates, which are both correct. I wonder if your pytorch training code is available or any suggestions to let the pytorch training work are welcome. 

Thanks!

**Comment by shayxurui** [comment]
hello:
 i wanna to retrain this network,too. so could u share ur training code?
thks,my email is 17212010044@fudan.edu.cn

**Comment by 1adrianb** [comment]
Unfortunatelly we have no imediate plans on preparing the pytorch training code, however if you correctly converted the lua training code you should be able to match or get close to the reported results.
